### PR TITLE
chore: re-architect quote slice (wip)

### PR DESCRIPTION
**Pull request by 0xApotheosis**
## Description

Re-architect the quote API to store response data in the slice in a more normalised way.

Now we only store:

- The selected swapper
- A list of all quote responses (including inputOutputRatio), which is now computed inside the RTK query endpoint itself, so consumers no longer need be concerned with computation or setting slice state

Also moves us towards end-state API design where all quotes come back in one response (though we lose swapper-specific loading state, 😔).

## Pull Request Type

- [ ] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [x] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [ ] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

N/A

## Risk

Small - contained within a feature flag that is disabled in prod.

## Testing

With the multi-hop feature flag on, quotes should display as expected.

### Engineering

Inspect the `tradeQuoteSlice` using the Redux dev tools and ensure it looks as expected/sane.

### Operations

☝️

## Screenshots (if applicable)

N/A

### CODE REVIEW

**Comment by 0xApotheosis** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #4821** <a href="https://app.graphite.dev/github/pr/shapeshift/web/4821" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/shapeshift/web/4821?utm_source=stack-comment).

File: src/state/slices/tradeQuoteSlice/tradeQuoteSlice.ts [hunk 1]
```diff
@@ -1,17 +1,16 @@
...
-import type { SwapErrorRight, SwapperName, TradeQuote } from 'lib/swapper/api'
+import type { SwapperName } from 'lib/swapper/api'
+import type { ApiQuote } from 'state/apis/swappers'
 
 export type TradeQuoteSliceState = {
   swapperName: SwapperName | undefined
-  quote: TradeQuote | undefined
-  error: SwapErrorRight | undefined
+  quotes: ApiQuote[]
 }
```
Comment (0xApotheosis) [inline src/state/slices/tradeQuoteSlice/tradeQuoteSlice.ts:9]: The crux of this PR.

File: src/state/apis/swappers/swappersApi.ts [hunk 2]
```diff
@@ -1,9 +1,68 @@
...
+                swapperName: result.swapperName,
+              })
+            : -Infinity
+          return { quote, error, inputOutputRatio, swapperName: result.swapperName }
+        })
+        const orderedQuotes: ApiQuote[] = orderBy(
+          quotesWithInputOutputRatios,
+          ['inputOutputRatio', 'swapperName'],
+          ['asc', 'asc'],
+        )
```
Comment (0xApotheosis) [inline src/state/apis/swappers/swappersApi.ts:56]: This logic moved into the API to more closely represent the ShapeShift API end-state.

File: src/state/apis/swappers/swappersApi.ts [hunk 3]
```diff
@@ -1,9 +1,68 @@
...
+
+        const bestQuote = orderedQuotes[0]
+        dispatch(tradeQuoteSlice.actions.setSwapperName(bestQuote.swapperName))
+        dispatch(tradeQuoteSlice.actions.setQuotes(orderedQuotes))
```
Comment (0xApotheosis) [inline src/state/apis/swappers/swappersApi.ts:60]: Update the slice directly from the RTK endpoint.

File: src/components/MultiHopTrade/MultiHopTrade.tsx [hunk 4]
```diff
@@ -76,17 +81,10 @@ export const MultiHopTrade = (props: CardProps) => {
...
-    [selectedQuote?.data],
-  )
-  const errorData = useMemo(
-    () => (selectedQuote?.data?.isErr() ? selectedQuote.data.unwrapErr() : undefined),
-    [selectedQuote?.data],
-  )
+  const selectedSwapperName = useAppSelector(selectSelectedSwapperName)
+  const selectedQuote = useAppSelector(selectSelectedQuote)
+  const selectedQuoteError = useAppSelector(selectSelectedQuoteError)
+  const isLoading = false // fixme
```
Comment (woodenfurniture) [inline src/components/MultiHopTrade/MultiHopTrade.tsx:87]: is this fixme in scope for this PR?

File: src/components/MultiHopTrade/MultiHopTrade.tsx [hunk 5]
```diff
@@ -116,18 +114,23 @@ export const MultiHopTrade = (props: CardProps) => {
...
   }, [selectedQuoteStatus.validationErrors])
 
   return (
     <MessageOverlay show={isKeplr} title={overlayTitle}>
-      {quoteData && <TradeConfirm tradeQuote={quoteData} swapperName={selectedQuote.swapperName} />}
+      {selectedQuote && (
+        <TradeConfirm
+          tradeQuote={selectedQuote}
+          swapperName={selectedSwapperName ?? SwapperName.Thorchain}
+        /> // fixme
```
Comment (woodenfurniture) [inline src/components/MultiHopTrade/MultiHopTrade.tsx:132]: what about this guy?

File: src/components/MultiHopTrade/components/TradeQuotes/TradeQuote.tsx [hunk 6]
```diff
@@ -1,15 +1,13 @@
...
-import type { Result } from '@sniptt/monads/build'
 import { useCallback, useMemo } from 'react'
 import { FaGasPump } from 'react-icons/fa'
 import { useTranslate } from 'react-polyglot'
 import { Amount } from 'components/Amount/Amount'
 import { RawText } from 'components/Text'
 import { useIsTradingActive } from 'components/Trade/hooks/useIsTradingActive'
 import { bnOrZero } from 'lib/bignumber/bignumber'
-import type { SwapErrorRight } from 'lib/swapper/api'
+import type { SwapErrorRight, TradeQuote as TradeQuoteType } from 'lib/swapper/api'
```
Comment (woodenfurniture) [inline src/components/MultiHopTrade/components/TradeQuotes/TradeQuote.tsx:9]: this might appear when you rebase, but in case its not
```suggestion
import type { SwapErrorRight, TradeQuote2 as TradeQuoteType } from 'lib/swapper/api'
```

File: src/components/MultiHopTrade/components/TradeQuotes/TradeQuotes.tsx [hunk 7]
```diff
@@ -1,24 +1,33 @@
...
+  const bestQuoteData = sortedQuotes?.[0]
 
-  const quotes = sortedQuotes.map((quoteData, i) => {
-    const { data, swapperName } = quoteData
-    const quote = data?.isOk() ? data.unwrap() : undefined
+  const quotes = sortedQuotes?.map((apiQuote, i) => {
+    const { quote, swapperName, inputOutputRatio, error } = apiQuote
+
+    const quoteDataShim: TradeQuoteLoadedProps['quoteData'] = {
+      isLoading: false,
```
Comment (woodenfurniture) [inline src/components/MultiHopTrade/components/TradeQuotes/TradeQuotes.tsx:22]: loading state will need to be plumbed through from the api

Comment (woodenfurniture) [reply src/components/MultiHopTrade/components/TradeQuotes/TradeQuotes.tsx:22]: you could potentially store the loading state in the store, not sure if that breaks our idea of 1 source of truth though

File: src/components/MultiHopTrade/hooks/useGetTradeQuotes.tsx [hunk 8]
```diff
@@ -78,61 +70,6 @@ export const useGetTradeQuotes = () => {
...
-  const error = bestQuote?.data && bestQuote.data.isErr() ? bestQuote.data.unwrapErr() : undefined
-  dispatch(tradeQuoteSlice.actions.setSwapperName(bestQuote?.swapperName))
-  dispatch(tradeQuoteSlice.actions.setQuote(quote))
-  dispatch(tradeQuoteSlice.actions.setError(error))
-  return {
-    sortedQuotes,
-    selectedQuote: sortedQuotes[0],
-  }
+  const quotes = useGetTradeQuoteQuery(debouncedTradeQuoteInput)
+  console.log('xxx quotes', quotes)
```
Comment (woodenfurniture) [inline src/components/MultiHopTrade/hooks/useGetTradeQuotes.tsx:74]:
```suggestion
```
